counterpartylib/lib/api.py

```diff
@@ -382,21 +384,34 @@ def run(self):
 
 class APIServer(threading.Thread):
     """Handle JSON-RPC API calls."""
-    def __init__(self):
+    def __init__(self, db=None):
+        self.db = db
```

Comment:
can you explain the reason for adding `self.db` vs the way it was done before? (I will probably find out as I review, but any pointers here are helpful...e.g. if it is a design decision, vs something you had to do due to a runtime issue with the picopayments enhancements)